File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int number) {
        int original=number;
        int reversed=0;
        while(number>0){
            int remainder=number%10;
            reversed=(reversed*10)+remainder;
            number=number/10;
        }
        if(reversed==original){
            return true;
        }return false;
}
public static void main(String args[]){
    Solution s=new Solution();
    System.out.println(s.isPalindrome(121));
    System.out.println(s.isPalindrome(-121));
    System.out.println(s.isPalindrome(10));
}
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        #rejects negative numbers beforehand
        if(x<0):
            return False

        original=x
        reversed=0
        while(x>0):
            remainder=x%10
            reversed=(reversed*10)+remainder
            x=x//10
        if (original==reversed):
            return True
        return False
        